src/Grpc.StatusProto/README.md

```diff
@@ -0,0 +1,256 @@
+
+If the client knows what details to expect for a specific error code then it can use the extension methods the explicitly extract the know type from the status details. 
+
+Example:
+```C#
+void PrintError(RpcException ex)
+{
+    // Get the status from the RpcException
+    Google.Rpc.Status? rpcStatus = ex.GetRpcStatus(); // Extension method
+
+    if (rpcStatus != null)
+    {
+        Console.WriteLine($"Google.Rpc Status: Code: {rpcStatus.Code}, Message: {rpcStatus.Message}");
+
+        // Try and get the ErrorInfo from the details
+        ErrorInfo? errorInfo = rpcStatus.GetErrorInfo(); // Extension method
+        if (errorInfo != null)
+        {
+            Console.WriteLine($"\tErrorInfo: Reason: {errorInfo.Reason}, Domain: {errorInfo.Domain}");
+            foreach (var md in errorInfo.Metadata)
+            {
+                Console.WriteLine($"\tKey: {md.Key}, Value: {md.Value}");
+            }
+        }
+        // etc ...
+    }
+}
+```
+
+Alternatively the client can walk the list of details contained in the status, processing
+each object. The class `DetailsTypesRegistry` provides an `Unpack` method to decode the `Any` messages in the status details. C#'s switch pattern matching can be used to process the details.
+
+Example:
+```C#
+void PrintStatusDetails(RpcException ex)
+{
+    // Get the status from the RpcException
+    Google.Rpc.Status? rpcStatus = ex.GetRpcStatus(); // Extension method
+
+    if (rpcStatus != null)
+    {
+        // Decode each "Any" item from the details in turn
+        foreach (Any any in rpcStatus.Details)
+        {
+            // decode the message if it is one of those expected to be in the
+            // status details
+            IMessage msg = DetailsTypesRegistry.Unpack(any); 
+
+            switch (msg)
+            {
+                case null:
+                    // ignore
+                    Console.WriteLine("Unknown message type in Details");
+                    break;
+
+                case ErrorInfo errorInfo:
+                    Console.WriteLine($"ErrorInfo: Reason: {errorInfo.Reason}, Domain: {errorInfo.Domain}");
+                    foreach (var md in errorInfo.Metadata)
+                    {
+                        Console.WriteLine($"\tKey: {md.Key}, Value: {md.Value}");
+                    }
+                    break;
+
+                case BadRequest badRequest:
+                    Console.WriteLine("BadRequest:");
+                    foreach (BadRequest.Types.FieldViolation fv in badRequest.FieldViolations)
+                    {
+                        Console.WriteLine($"\tField: {fv.Field}, Description: {fv.Description}");
+                    }
+                    break;
+
+                // Other cases handled here ...
+            }
+        }
+    }
+
+```
+
+## Returning errors within gRPC streams
```

Comment:
I didn't know about the streaming use case with per-message error details. Where did you find that?